app/main.py

```diff
@@ -2,0 +1,49 @@
+class Animal:
+    def __init__(
+            self,
+            name: str,
+            appetite: int,
+            is_hungry: bool = True) -> None:
+        self.name = name
+        self.appetite = appetite
+        self.is_hungry = is_hungry
+
+    def print_name(self) -> None:
+        print(f"Hello, I'm {self.name}")
+
+    def feed(self) -> int:
+        if self.is_hungry:
+            print(f"Eating {self.appetite} food points...")
+            self.is_hungry = False
+            return self.appetite
+        return 0
+
+
+class Cat(Animal):
+    def __init__(self,
+                 name: str,
+                 is_hungry: bool = True) -> None:
+        super().__init__(name=name, appetite=3)
+        self.name = name
+        self.is_hungry = is_hungry
+
+    @staticmethod
+    def catch_mouse() -> None:
+        print("The hunt began!")
+
+
+class Dog(Animal):
+    def __init__(self,
+                 name: str,
+                 is_hungry: bool = True) -> None:
+        super().__init__(name=name, appetite=7)
+        self.name = name
+        self.is_hungry = is_hungry
```

Comment:
Similar to the `Cat` class, the assignments `self.name = name` and `self.is_hungry = is_hungry` in the `Dog` class constructor are redundant. These are already set by the `super().__init__` call, so you can remove these lines.